x-pack/plugins/viz_editor/public/pseudo_plugins/indexpattern_datasource_plugin/to_expression.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Query, WhereOperation } from '../../../common';
+import { VisModel } from '../../common/lib';
+
+function isRawDocumentQuery(query: Query) {
+  return query.select.every(selectOperation => selectOperation.operation === 'column');
+}
+
+function queryToEsAggsConfigs(query: Query): any {
+  return [...query.select].reverse().map((selectOperation, index) => {
```

Comment:
I've found it useful to define these kinds of operationName -> object mappings as separate types, such as in `select_builder.ts`- maybe you can use the same pattern here?